File: leetcode/1400-1499/1455. Check If a Word Occurs As a Prefix of Any Word in a Sentence/1455.cs
public class Solution
{
    public int IsPrefixOfWord(string sentence, string searchWord)
    {
        string[] words = sentence.Split(" ");
        int searchLength = searchWord.Length;
        for (int i = 0; i < words.Length; i++)
        {
            string word = words[i];
            int wordLength = word.Length;
            if (wordLength >= searchLength)
            {
                bool isPrefix = true;
                for (int j = 0; j < searchLength; j++)
                {
                    if (word[j] != searchWord[j])
                    {
                        isPrefix = false;
                        break;
                    }
                }
                if (isPrefix)
                    return i + 1;
            }
        }
        return -1;
    }
}

File: leetcode/1400-1499/1455. Check If a Word Occurs As a Prefix of Any Word in a Sentence/1455.py
class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        words = sentence.split(" ")

        for i, word in enumerate(words):
            # if len(word)>=len(searchWord) and searchWord==word[:len(searchWord)]:
            #     return i+1
            if str.startswith(word, searchWord):
                return i + 1
            
        return -1
# TC: O(n*m)
# SC: O(n)